src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -97,8 +97,6 @@ private Task SetRoutingAndContinue(HttpContext httpContext, EndpointSelectorCont
             {
                 // Set the endpoint feature only on success. This means we won't overwrite any
                 // existing state for related features unless we did something.
-                SetFeatures(httpContext, feature);
-
                 Log.MatchSuccess(_logger, feature);
```

Comment:
This isn't right

Reply:
Well, it used to be lel

Reply:
Yah, to keep the same behaviour you'd have to save and restore endpoint and rvd. 

OR 

we could dicuss this more and see what the right behaviour is when you ahve multiple routes. I know Jimmy is particularly passionate about this topic, but I made us just do something simple.